### PR TITLE
refactor code consistency #112

### PR DESCRIPTION
**Pull request by AntonioliBenjamin**
hey @ipapandinas 

I hope that all the elements are as you wished, 
don't hesitate if you need to change something.

### CODE REVIEW

File: src/eventHandlers/nftOperations.ts [hunk 1]
```diff
@@ -66,3 +66,15 @@ export const nftOperationEntityHandler = async (
   }
   await nftOperationRecord.save()
 }
+
+export enum NFTOperation {
```
Comment (ipapandinas) [inline src/eventHandlers/nftOperations.ts:70]: We should use lower case for values and also add additional auction's operations: 

```
export enum NFTOperation {
  Create = "create",
  Burn = "burn",
  Transfer = "transfer",
  Delegate = "delegate",
  Undelegate = "undelegate",
  SetRoyalty = "setRoyalty",
  Sell = "sell",
  List = "list",
  Unlist = "unlist",
  CompleteAuction = "completeAuction",
  BuyItNowAuction = "buyItNowAuction",
  CreateAuction = "createAuction",
  AddBid = "addBid",
  RemoveBid = "removeBid",
}
```

**Review by ipapandinas — COMMENTED** [review]
Nice work @AntonioliBenjamin 🤩
Now it is time to spread and use the new enum NFTOperation all over the code:

- for the switch/case inside `nftOperationEntityHandler`

```javascript
// BEFORE
case "create":
      nftOperationRecord.to = record.owner
      break
case "burn":
  nftOperationRecord.to = null
  break

// NOW
case NFTOperation.Create:
      nftOperationRecord.to = record.owner
      break
case NFTOperation.Burn:
  nftOperationRecord.to = null
  break

...etc
```
- each time `nftOperationEntityHandler` is used

```javascript
// BEFORE
await nftOperationEntityHandler(record, null, commonEventData, "create")

// NOW
await nftOperationEntityHandler(record, null, commonEventData, NFTOperation.Create)

...etc
```

Then it will be perfect!

**Comment by AntonioliBenjamin** [comment]
hi ! @ipapandinas 

in auctions.ts line 86 i see `await nftOperationEntityHandler(nftRecord, record.creator, commonEventData, "cancelAuction")`
in nfts.ts line 185 i see ` await nftOperationEntityHandler(nftRecord, collectionRecord.owner, commonEventData, "addToCollection")`

i need to add `cancelAuction `& `addToCollection`  to the enum ?

**Comment by ipapandinas** [comment]
good point, we have to add both of them

**Comment by AntonioliBenjamin** [comment]
@ipapandinas 

Done, let me know if something is wrong  :)

**Review by ipapandinas — COMMENTED** [review]
I just spotted one last thing. We can improve `typeOfTransaction` type in `nftOperationEntityHandler`'s parameters. Instead of **string**, we can use **NFTOperation** now:

```javascript
// BEFORE
export const nftOperationEntityHandler = async (
  record: NftEntity,
  oldOwner: string,
  commonEventData: CommonEventData,
  typeOfTransaction: string,
  args?: any[],
): Promise<void> => {
...
}

// NOW
export const nftOperationEntityHandler = async (
  record: NftEntity,
  oldOwner: string,
  commonEventData: CommonEventData,
  typeOfTransaction: NFTOperation,
  args?: any[],
): Promise<void> => {
...
}
```

typeOfTransaction: NFTOperation,

**Comment by AntonioliBenjamin** [comment]
@ipapandinas 

Done, now in nfts.ts line 100 we have a problem.
replace `typeOfTransaction` by : `NFTOperation.Delegate` or  `NFTOperation.Undelegate` ?

then it's over :D

**Comment by AntonioliBenjamin** [comment]
hi @ipapandinas 

regarding nfts.ts file line 101,

```
  const typeOfTransaction = record.delegatee ? "delegate" : "undelegate"
  await nftOperationEntityHandler(record, record.owner, commonEventData, typeOfTransaction)
```

we need to asing  `NFTOperation.Delegate || NFTOperation.Undelegate` in place of  `typeOfTransaction` or i don't change this one ?

typeOfTransaction type in nftOperationEntityHandler's parameters is done

**Review by ipapandinas — COMMENTED** [review]
Amazing, everything is perfect 🤩
I merge it!

**Pull request merged**